nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -85,8 +85,13 @@ let
 
   startScript = cfg:
     ''
-      mkdir -p -m 0755 "$root/etc" "$root/var/lib"
-      mkdir -p -m 0700 "$root/var/lib/private" "$root/root" /run/nixos-containers
+      # Declare root explicitly to avoid shellcheck warnings, it comes from the env
+      declare root
+
+      mkdir -p "$root/etc" "$root/var/lib"
+      chmod 0755 "$root/etc" "$root/var/lib"
+      mkdir -p "$root/var/lib/private" "$root/root" /run/nixos-containers
+      chmod 0700 "$root/var/lib/private" "$root/root" /run/nixos-containers
```

Comment:
I am not sure what the layout is before this. If $root already exists, we could also drop the -p where possible.

Also I don't think your change is fixing the issue shellcheck is noting. The directories along the path still will have the same permissions as before and the ones created by -p still have the default umask permissions.

Reply:
This was pulled out of this PR.